tests/abortStatus/host/host.cpp

```diff
@@ -91,53 +85,48 @@ static void OcallAfterCrashThread(
     std::atomic<uint32_t>* thread_ready_count,
     std::atomic<bool>* is_enclave_crashed)
 {
-    oe_result_t result;
-    AbortStatusArgs args;
-    args.ret = -1;
-    args.thread_ready_count = thread_ready_count;
-    args.is_enclave_crashed = is_enclave_crashed;
-
-    result = oe_call_enclave(enclave, "TestOCallAfterAbort", &args);
+    int rval = 0;
+    void* _thread_ready_count = reinterpret_cast<void*>(thread_ready_count);
+    void* _is_enclave_crashed = reinterpret_cast<void*>(is_enclave_crashed);
+    oe_result_t result = test_ocall_after_abort(
+        enclave, &rval, &_thread_ready_count, &_is_enclave_crashed);
     OE_TEST(result == OE_OK);
-    OE_TEST(args.ret == 0);
+    OE_TEST(0 == rval);
 
     return;
 }
+
 // Test the regular abort case and un-handled hardware exception case in
 // single thread.
 static bool TestBasicAbort(const char* enclave_name)
 {
-    oe_result_t result;
-    oe_enclave_t* enclave = NULL;
-
     const uint32_t flags = oe_get_create_flags();
-    const char* function_names[] = {"RegularAbort",
-                                    "GenerateUnhandledHardwareException"};
 
-    for (uint32_t i = 0; i < OE_COUNTOF(function_names); i++)
+    std::pair<char const*, enc_fn> functions[] = {
+        std::make_pair("regular_abort", regular_abort),
+        std::make_pair("generate_unhandled_hardware_exception",
+                       generate_unhandled_hardware_exception),
+    };
+    
+    for (uint32_t i = 0; i < OE_COUNTOF(functions); ++i)
     {
-        if ((result = oe_create_enclave(
-                 enclave_name,
-                 OE_ENCLAVE_TYPE_SGX,
-                 flags,
-                 NULL,
-                 0,
-                 NULL,
-                 0,
-                 &enclave)) != OE_OK)
+        oe_enclave_t* enclave = NULL;
+        oe_result_t result = oe_create_abortStatus_enclave(
+            enclave_name, OE_ENCLAVE_TYPE_SGX, flags, NULL, 0, &enclave);
+        if (OE_OK != result)
         {
-            oe_put_err("oe_create_enclave(): result=%u", result);
+            oe_put_err("oe_create_abortStatus_enclave(): result=%u", result);
             return false;
         }
 
         // Skip the last test for simulation mode.
         if ((flags & OE_ENCLAVE_FLAG_SIMULATE) == 0 ||
-            (i != OE_COUNTOF(function_names) - 1))
+            (i != OE_COUNTOF(functions) - 1))
         {
-            TestAbortStatus(enclave, function_names[i]);
+            TestAbortStatus(enclave, functions[i].first, functions[i].second);
         }
 
-        if ((result = oe_terminate_enclave(enclave)) != OE_OK)
+        if ((result = oe_terminate_enclave(enclave)) != OE_MEMORY_LEAK)
```

Comment:
This likely needs to be special cased for Release builds since memory leak detection is not enabled in release builds.